tests/net/npf/src/main.c

```diff
@@ -307,4 +342,156 @@ ZTEST(net_pkt_filter_test_suite, test_npf_address_mask)
+				 &ipv4_address_list[it], sizeof(struct in_addr));
+		zassert_false(net_pkt_filter_ip_recv_ok(pkt_v4), "");
+	}
+
+	/* And one not listed */
+	memcpy((struct in_addr *)NET_IPV4_HDR(pkt_v4)->src,
+				 &bad_addr, sizeof(struct in_addr));
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v4), "");
+
+	zassert_true(npf_remove_all_ipv4_recv_rules(), "");
+	net_pkt_unref(pkt_v6);
+	net_pkt_unref(pkt_v4);
+}
+
+static NPF_IP_SRC_ADDR_ALLOWLIST(allowlist_ipv6_src_addr, (void *)ipv6_address_list,
+					ARRAY_SIZE(ipv6_address_list), AF_INET6);
+static NPF_IP_SRC_ADDR_BLOCKLIST(blocklist_ipv6_src_addr, (void *)ipv6_address_list,
+					ARRAY_SIZE(ipv6_address_list), AF_INET6);
+
+static NPF_RULE(ipv6_allowlist, NET_OK, allowlist_ipv6_src_addr);
+static NPF_RULE(ipv6_blocklist, NET_OK, blocklist_ipv6_src_addr);
+
+static void test_npf_ipv6_address_filtering(void)
+{
+	struct in6_addr dst = { { { 0xfe, 0x80, 0x43, 0xb8, 0, 0, 0, 0, 0, 0, 0,
+								0xf2, 0xaa, 0x29, 0x02, 0x04 } } };
+	struct in6_addr bad_addr = { { { 0x20, 0x01, 0x0d, 0xb8, 8, 0, 0, 0, 0,
+								0, 0, 0, 0, 0, 0, 0x1 } } };
+	struct net_pkt *pkt_v6 = build_test_ip_pkt(&ipv6_address_list[0],
+							&dst, AF_INET6, &dummy_iface_a);
+
+	struct net_pkt *pkt_v4 = build_test_ip_pkt(&ipv4_address_list[0],
+							&ipv4_address_list[1], AF_INET, &dummy_iface_a);
+
+	/* make sure pkt is initially accepted */
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v4), "");
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+
+	/* validate allowlist */
+	npf_insert_ipv6_recv_rule(&ipv6_allowlist);
+
+	for (int it = 0; it < ARRAY_SIZE(ipv6_address_list); it++) {
+		memcpy((struct in6_addr *)NET_IPV6_HDR(pkt_v6)->src,
+				 &ipv6_address_list[it], sizeof(struct in6_addr));
+		zassert_true(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+	}
+
+	/* And one not listed */
+	memcpy((struct in6_addr *)NET_IPV6_HDR(pkt_v6)->src,
+				 &bad_addr, sizeof(struct in6_addr));
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v4), "");
+	zassert_false(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+
+	/* Prepare new test */
+	zassert_true(npf_remove_all_ipv6_recv_rules(), "");
+
+	/* make sure pkt is initially accepted */
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v4), "");
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+
+	/* validate blocklist */
+	npf_insert_ipv6_recv_rule(&ipv6_blocklist);
+
+	for (int it = 0; it < ARRAY_SIZE(ipv6_address_list); it++) {
+		memcpy((struct in6_addr *)NET_IPV6_HDR(pkt_v6)->src,
+				 &ipv6_address_list[it], sizeof(struct in6_addr));
+		zassert_false(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+	}
+
+	/* And one not listed */
+	memcpy((struct in6_addr *)NET_IPV4_HDR(pkt_v6)->src,
+				 &bad_addr, sizeof(struct in6_addr));
+	zassert_true(net_pkt_filter_ip_recv_ok(pkt_v6), "");
+
+	zassert_true(npf_remove_all_ipv6_recv_rules(), "");
+	net_pkt_unref(pkt_v6);
+	net_pkt_unref(pkt_v4);
+}
+
+ZTEST(net_pkt_filter_test_suite, test_npf_ip_filter)
+{
+	test_npf_ipv4_address_filtering();
```

Comment:
IMO we should just split IPv4/IPv6 testing into two separate test cases.

Reply:
I fully agree, however file author did it another way:
https://github.com/nrfconnect/sdk-zephyr/blob/01e2e4c9ce8294d74032a2d33e0dcb1b74802568/tests/net/npf/src/main.c#L304-L308
Should I change my TC and this above as well?

Reply:
I don't think there's a need to modify existing tests, just update the new ones. Unless you really want to.

Reply:
I think that common approach allow us keep clear code.

Reply:
OK, I see that I cannot change this TC
https://github.com/nrfconnect/sdk-zephyr/blob/01e2e4c9ce8294d74032a2d33e0dcb1b74802568/tests/net/npf/src/main.c#L287
So I update only my part.